modular_skyrat/modules/antag_optin/code/job.dm

```diff
@@ -0,0 +1,69 @@
+/datum/job
+	/// The minimum antag opt-in any holder of this job must use. If null, will defer to the mind's opt in level.
+	var/minimum_opt_in_level
+	/// Can this job be targetted as a heretic sacrifice target?
+	var/heretic_sac_target
+	/// Is this job targettable by contractors?
+	var/contractable
+
+/// Updates [minimum_opt_in_level] [heretic_sac_target] and [contractable].
+/datum/job/proc/update_opt_in_vars()
+	if (!CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		if (isnull(minimum_opt_in_level))
+			minimum_opt_in_level = get_initial_opt_in_level()
+		if (isnull(heretic_sac_target))
+			heretic_sac_target = initialize_heretic_target_status()
+		if (isnull(contractable))
+			contractable = initialize_contractable_status()
```

Comment:
I feel like those variables could also be added to the job configs, possibly. Maybe not now, but something to think about in the future.